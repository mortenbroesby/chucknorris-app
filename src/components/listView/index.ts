import Vue from "vue";
import { Component, Prop } from "vue-property-decorator";
import { List } from "../../interfaces";

import template from "./listView.vue";
import "./listView.scss";

@Component({
  mixins: [template],
  components: {}
})
export default class ListView extends Vue {
  /*************************************************/
  /* EXTERNAL PROPERTIES */
  /*************************************************/
  @Prop({ default: [] })
  list: List[];

  @Prop({ default: "" })
  title: string;

  @Prop({ default: false })
  isVisible: boolean;

  @Prop({ default: "favorite" })
  buttonIcon: string;

  /*************************************************/
  /* METHODS */
  /*************************************************/
  isClicked(item: any) {
    this.$emit("click", item);
  }
}
