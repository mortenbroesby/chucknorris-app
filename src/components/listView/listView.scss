@import "../../styles/global.scss";

.listView {
  margin-top: 10px;
  margin-bottom: 60px;

  .title {
    padding: 0;
    margin: 0;
    margin-bottom: 20px;
    text-transform: uppercase;
  }

  .listView--item {
    display: flex;
    flex-flow: row wrap;
    align-items: center;
    justify-content: center;
    min-height: 90px;

    &:not(:last-child) {
      border-top: 1px solid color("grey");
    }

    &:last-child {
      border-top: 1px solid color("grey");
      border-bottom: 1px solid color("grey");
    }

    .listView--message {
      flex: 1 80%;
      line-height: 1.5;
      font-size: 13px;
    }

    .listView--button {
      border: none;
      outline: none;
      background: none;

      width: 40px;
      height: 40px;
      margin-left: 30px;
      display: flex;
      flex-flow: row wrap;
      align-items: center;
      justify-content: center;

      & > * {
        vertical-align: middle;
        line-height: 0.5;
        color: black;
      }
    }
  }
}