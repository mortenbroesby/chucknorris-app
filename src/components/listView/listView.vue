<template>
  <div class="listView" v-if="isVisible">
    <h5 class="title">{{ title }}</h5>
    <div class="listView--item"
      v-for="(item, index) in list"
      :key="index">
      <p class="listView--message" v-html="item.message"></p>
      <button
        class="listView--button"
        v-on:click="isClicked(item)">
        <i class="material-icons">{{ buttonIcon }}</i>
      </button>
    </div>
  </div>
</template>
