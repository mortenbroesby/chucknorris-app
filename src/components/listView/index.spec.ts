import { mount, createLocalVue } from "@vue/test-utils";

import listView from "./index";

describe("components/listView", () => {
  const localVue = createLocalVue();

  test("should have correct default rendering", done => {
    const wrapper = mount(listView, {
      localVue,
      attachToDocument: true,
      propsData: {
        list: [],
        title: "Title 1",
        isVisible: true,
        buttonIcon: ""
      }
    });

    wrapper.vm.$nextTick(() => {
      expect(wrapper.isVueInstance()).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
      wrapper.destroy();
      done();
    });
  });

  test("should have correct rendering with actual data", done => {
    const list = [
      { message: "Message 1" },
      { message: "Message 2" },
    ];

    const wrapper = mount(listView, {
      localVue,
      attachToDocument: true,
      propsData: {
        list: list,
        title: "Title 2",
        isVisible: true,
        buttonIcon: "favorite"
      }
    });

    wrapper.vm.$nextTick(() => {
      expect(wrapper.isVueInstance()).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
      wrapper.destroy();
      done();
    });
  });
});
