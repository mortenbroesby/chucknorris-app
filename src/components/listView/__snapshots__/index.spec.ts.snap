// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/listView should have correct default rendering 1`] = `
<div
  class="listView"
>
  <h5
    class="title"
  >
    Title 1
  </h5>
   
</div>
`;

exports[`components/listView should have correct rendering with actual data 1`] = `
<div
  class="listView"
>
  <h5
    class="title"
  >
    Title 2
  </h5>
   
  <div
    class="listView--item"
  >
    <p
      class="listView--message"
    >
      Message 1
    </p>
     
    <button
      class="listView--button"
    >
      <i
        class="material-icons"
      >
        favorite
      </i>
    </button>
  </div>
  <div
    class="listView--item"
  >
    <p
      class="listView--message"
    >
      Message 2
    </p>
     
    <button
      class="listView--button"
    >
      <i
        class="material-icons"
      >
        favorite
      </i>
    </button>
  </div>
</div>
`;
